pages/api/auth/[...nextauth].ts

```diff
@@ -2,27 +2,39 @@ import NextAuth, { AuthOptions } from 'next-auth';
 import { JWT } from 'next-auth/jwt';
 import { OAuthConfig } from 'next-auth/providers';
 
-import { SessionWithToken } from '@orchestrator-ui/orchestrator-ui-components';
+import {
+    WfoSession,
+    WfoUserProfile,
+} from '@orchestrator-ui/orchestrator-ui-components';
 
 const token_endpoint_auth_method = process.env.NEXTAUTH_CLIENT_SECRET
     ? 'client_secret_basic'
     : 'none';
 
-export interface WfoProfile extends Record<string, string> {}
-
 const authActive = process.env.AUTH_ACTIVE?.toLowerCase() != 'false';
-const WfoProvider: OAuthConfig<WfoProfile> = {
+const wfoProvider: OAuthConfig<WfoUserProfile> = {
     id: process.env.NEXTAUTH_ID || '',
     name: process.env.NEXTAUTH_ID || '',
+    type: 'oauth',
     clientId: process.env.NEXTAUTH_CLIENT_ID || '',
     clientSecret: process.env.NEXTAUTH_CLIENT_SECRET || undefined,
-    wellKnown: `${
-        process.env.NEXTAUTH_ISSUER || ''
-    }/.well-known/openid-configuration`,
-    type: 'oauth',
+    wellKnown:
+        process.env.NEXTAUTH_WELL_KNOWN_OVERRIDE ??
+        `${process.env.NEXTAUTH_ISSUER || ''}/.well-known/openid-configuration`,
     authorization: { params: { scope: 'openid profile' } },
-    checks: ['pkce', 'state'],
     idToken: true,
+    checks: ['pkce', 'state'],
+    userinfo: {
```

Comment:
Is this the extra call to userinfo you have to do because NextAuth doesn't do it? I don't know if this is the place but it would be good to document this behavior and why you're doing it somewhere. So there is some record of the rabit hole you went down